### ISSUE TITLE
Customer group promotion condition not encoded correctly with uuid strategy

### ISSUE THREAD

**Issue by jacobfrantz1** [vendure-ecommerce/vendure#2019]
**Describe the bug**
When using uuid entity id strategy, the customer group promotion condition does not work. I think it has to do with encoding the condition in the create and update promotion mutations, as manually changing the database to remove the escaped quotes makes the promotion work as intended.

When trying to track down the issue in the code, a print inside the idsAreEqual function revealed the customerGroupId was encoded with an extra set of quotation marks
e.g. 53dc7cc8-2808-4f2c-90f5-aa4e8725f9fd != "53dc7cc8-2808-4f2c-90f5-aa4e8725f9fd"

**To Reproduce**
1. Use the UUID Entity Id Strategy
2. Create a Customer Group
3. Create a Promotion with the Customer Group Condition
4. Refresh Page, Customer group is now empty. Promotion is not applied to customer group.

**Expected behavior**
The promotion should apply to the customer group after creating or updating the promotion via 

**Environment (please complete the following information):**
 - @vendure/core version: 1.9.2
 - Nodejs version: 16.18.1
 - Database: Postgresql 14.5

**Comment by jacobfrantz1** [comment]
This issue is still a problem, I ran into it today when adding a promotion, and it got worse in v2 with the admin-ui not able to set customer group condition with both uuid strategy and autoincrement id strategy.

I've tracked down the root of the issue, as well as a new issue introduced in v2.

I'll do a pull request in the next couple of days or so fixing this.

As for the new issue, the customer-group-form-input emits the group object instead of the id as of v2, and that doesn't work for the customer group promotion condition or another configurable operation I use in my server, as they expect ID/Array of IDs instead of Group/Array of Groups. This is stored in the db as "\"[Object object]\"".

This root issue I ran into is due to the configurable-operation-codec.

TLDR: ConfigurableOperationCodec's decodeConfigurableOperationIds shouldn't JSON.stringify the id if it is a string, and maybe shouldn't JSON.stringify the id at all (what I'm leaning towards), as it is JSON.stringified in the encodeConfigurableOperationIds function.

With the autoincrement id strategy, in decodeConfigurableOperationIds, the id is decoded to a number and JSON.stringified to "1" for example. In encodeConfigurableOperationIds, the id is encoded, a no-op as it was stringified earlier, and JSON.stringified to "\"1\"".

However with the uuid-strategy, the id decoding is a no-op, and it is JSON.stringified to "\"XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX\"" in the decode step, and stored on the database with an extra set of quotes. When it is encoded later and sent to the admin it looks like "\\\"XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX\\\"". The backend is expecting the id to not have the extra set of quotes for id comparison when checking promotions. The admin-ui is also not expecting the extra quotes.
